frontend/src/components/object/ObjectTableDeleted.vue

```diff
@@ -0,0 +1,369 @@
+      data-key="id"
+      class="p-datatable-sm"
+      responsive-layout="scroll"
+      :rows="10"
+      :rows-per-page-options="[10, 20, 50]"
+      sort-field="updatedAt"
+      :sort-order="-1"
+      :global-filter-fields="['name']"
+      :first="first"
+      update:filters
+      update:page
+      @page="onPage($event)"
+      @sort="onSort($event)"
+      @filter="onFilter($event)"
+    >
+      <template #header>
+        <div class="flex justify-content-end">
+          <span class="ml-4">
+            <InputText
+              v-model="filters.name.value"
+              class="pr-6"
+              placeholder="Search File Names"
+              @keyup.enter="loadLazyData"
+            />
+            <Button
+              v-show="filters.name.value"
+              v-tooltip.bottom="'Clear'"
+              class="overlap"
+              icon="pi pi-times"
+              outlined
+              aria-label="Clear"
+              @click="
+                () => {
+                  filters.name.value = undefined;
+                }
+              "
+            />
+          </span>
+          <Button
+            v-tooltip.bottom="'Search'"
+            class="ml-2"
+            icon="pi pi-search"
+            outlined
+            rounded
+            aria-label="Filter"
+            @click="onFilter()"
+          />
+        </div>
+      </template>
+      <template #empty>
+        <div
+          v-if="!loading"
+          class="flex justify-content-center"
+        >
+          <h4 class="py-5">Looks like you don't have any recently deleted files.</h4>
+        </div>
+      </template>
+      <template #loading>
+        <Spinner />
+      </template>
+      <Column
+        selection-mode="multiple"
+        header-style="width: 3rem"
+      />
+      <Column
+        field="name"
+        sortable
+        header="Name"
+        header-style="min-width: 25%"
+        body-class="truncate"
+      >
+        <template #body="{ data }">
+          <div>
+            <router-link :to="{ name: RouteNames.DETAIL_OBJECTS, query: { objectId: data.id } }">
+              <span v-tooltip.bottom="'View file details'">
+                {{ data.name }}
+              </span>
+            </router-link>
+          </div>
+        </template>
+      </Column>
+      <Column
+        field="location"
+        sortable
+        header="Original Location"
+      >
+        <template #body="{ data }">
+          <span v-tooltip.bottom="'The Original location of this fie'">
+            {{ data.location }}
+        </span>
+        </template>
+      </Column>
+      <Column
+        field="updatedAt"
+        header="Deleted date"
+        sortable
+      >
+        <template #body="{ data }">
+          {{ formatDateLong(data.updatedAt) }}
+        </template>
+      </Column>
+      <Column
+        header="Actions"
+        header-style="min-width: 150px"
+        header-class="header-right"
+        body-class="action-buttons"
+      >
```

Comment:
Action buttons are being wrapped to a new row if the "original location" column is too long.